src/main/kotlin/study/racingcar/repository/impl/RoundMemoryRepositoryImpl.kt

```diff
@@ -0,0 +1,27 @@
+package study.racingcar.repository.impl
+
+import study.racingcar.domain.Round
+import study.racingcar.repository.RoundRepository
+
+class RoundMemoryRepositoryImpl : RoundRepository {
+
+    private val repository: MutableList<Round> = mutableListOf()
+
+    override fun findAll(): List<Round> {
+        return repository
+    }
+
+    override fun findBy(number: Int): Round {
+        return repository.first { it.number == number }
+    }
+
+    override fun save(round: Round) {
+        val newRound = Round(round.number, round.cars.map { it.copy() })
+
+        repository.add(newRound)
+    }
+
+    override fun findLast(): Round {
+        return repository.maxByOrNull { it.number }!!
```

Comment:
max를 구할 수 없는 상황은 어떤 경우일까요?

!! operator를 사요앟면 NullPointerException이 발생할텐데, 해당 상황에 대해 알 수 있는 구체적인 예외를 던지면 좋을 것 같아요

```suggestion
        return repository.maxByOrNull { it.number } ?: throw ~~
```